src/liboslexec/llvm_util.cpp

```diff
@@ -585,6 +586,9 @@ LLVM_Util::~LLVM_Util()
     delete m_new_pass_manager;
     delete m_builder;
     delete m_llvm_debug_builder;
+    if (m_nvptx_target_machine != nullptr) {
+        delete m_nvptx_target_machine;
+    }
```

Comment:
I believe that this can be

```
    delete m_nvptx_target_machine;
```

C++ semantics say that delete itself checks that the pointer is non-null so you don't need to enclose it in yet another nullptr check.

Reply:
Thanks, I deleted the null check.